ocaml/lambda/translcore.ml

```diff
@@ -1341,6 +1372,8 @@ and transl_function0
     let arg_mode = transl_alloc_mode arg_mode in
     ((Curried {nlocal},
       [{ name = param;
+         var_uid = Ident.Tbl.find_opt (uids_of_vars cases) param
+            |> Option.value ~default:Uid.internal_not_actually_unique;
```

Comment:
Similarly, we can remember the uid instead of searching for it here, and then delete `uids_of_vars`.